src/index.ts

```diff
@@ -150,23 +174,30 @@ export class Client {
     });
 
     if (resp.status !== 200) {
+      let code;
       let message = resp.statusText;
       try {
-        let response = await resp.json();
-        message = response.message;
+        ({ code, message } = (await resp.json()) as APIError);
       } catch (_) {}
 
+      console.error(
+        `Search request failed due to an error. HTTP code: ${resp.status}, Code: ${code}, Message: ${message}`
```

Comment:
This `catch` block will only be entered if JSON parsing fails, which means `code` is always going to be `undefined`. You can remove it from this message.

Reply:
@JasonBerry logging an undefined code was somewhat intentional. The API call should always return it, so being able to see an API call didn't return it was my intention. However, absence of the `code` at all in the log message sounds like a good enough signal to me so will adjust.

Reply:
Urgh, my bad. Due to the unified diff I missed that the `catch` block when JSON parsing fails is empty.

Do we want to move this `console.error` below the `403` condition, just before we throw the generic error.

Reply:
Done. See commit message if you're interested for my previous rationale, but I've moved it below.